src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -177,4 +186,136 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
 
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, exam.getTitle())).build();
     }
+
+    /**
+     * Post /courses/:courseId/exams/:examId/students/:studentLogin : Add one single given user (based on the login) to the students of the exam so that the student can access the exam
+     *
+     * @param courseId     the id of the course
+     * @param examId       the id of the exam
+     * @param studentLogin the login of the user who should get student access
+     * @return empty ResponseEntity with status 200 (OK) or with status 404 (Not Found)
+     */
+    @PostMapping(value = "/courses/{courseId}/exams/{examId}/students/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> addStudentToExam(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable String studentLogin) {
+        log.debug("REST request to add {} as student to exam : {}", studentLogin, examId);
+        var course = courseService.findOne(courseId);
+        var instructorOrAdmin = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, instructorOrAdmin)) {
+            return forbidden();
+        }
+        var exam = examService.findOneWithRegisteredUsers(examId);
+        if (!course.equals(exam.getCourse())) {
+            return conflict();
+        }
```

Comment:
We could use `checkCourseAndExamAccess()` from `ExamAccessService` here.

Reply:
👍 I will change it in a follow-up PR